ui/src/app/shared/header/header.component.ts

```diff
@@ -281,8 +281,8 @@ export class JobsPaginatorIntl extends MatPaginatorIntl {
   getRangeLabel = (page: number, pageSize: number, length: number) => {
 
     let knownLength = null;
-
-    if (this.backendJobs.value.totalSize !== null) {
+    if (this.backendJobs.value.totalSize ||
```

Comment:
This looks like it's intended to check that `totalSize` "is defined", or at least, "is not undefined"? And that need a special case because `0` counts as false?

I guess I'm a little bit surprised there's not a neater way to express an "is defined" check (eg `this.backendJobs.value.totalSize !== undefined`)?

Reply:
Yes, that was the goal (and 0 is considered false).

I guess I'm just being extra safe since there was already a bug here. It's pretty common that a field would be null/undefined even though it's not documented as such. Your check should also have worked fine (`!== undefined`). If somehow a null showed up here (e.g. if a backend returned `totalResults: null`), this would catch it.

Reply:
It is enforced by the compiler, but there's a cast going on in the API  layer when we receive a JSON response. It would have been more correct if that code instead placed nulls for these missing properties (given the type signature of our models), but it doesn't do that currently.